src/data/list/perm.lean

```diff
@@ -690,6 +690,103 @@ theorem perm_iff_count {l₁ l₂ : list α} : l₁ ~ l₂ ↔ ∀ a, count a l
     by_cases b = a; simp [h] at H ⊢; assumption }
 end⟩
 
+lemma subperm.cons_right {α : Type*} {l l' : list α} (x : α) (h : l <+~ l') : l <+~ x :: l' :=
+h.trans (sublist_cons x l').subperm
+
+lemma subperm.cons_left {l₁ l₂ : list α} (h : l₁ <+~ l₂)
+  (x : α) (hx : count x l₁ < count x l₂) :
+  x :: l₁ <+~ l₂  :=
+begin
+  obtain ⟨l, hl, h⟩ := h,
+  induction h with lx ly y h IH lx ly y h IH generalizing l₁,
+  { simpa using hx },
+  { by_cases hxy : x = y,
+    { subst hxy,
+      rw subperm_cons,
+      exact hl.symm.subperm.trans h.subperm },
+    { rw count_cons_of_ne hxy at hx,
+      exact (IH hx hl).cons_right _ } },
+  { rw cons_perm_iff_perm_erase at hl,
+    by_cases hxy : x = y,
+    { subst hxy,
+      rw subperm_cons,
+      refine (subperm_cons_erase _ _).trans (IH _ hl.right),
+      rw count_erase_self,
+      refine (nat.pred_lt (count_pos.mpr hl.left).ne').trans_le _,
+      simpa [nat.lt_succ_iff] using hx, },
+    { rw count_cons_of_ne hxy at hx,
+      have : y :: x :: l₁.erase y ~ x :: l₁,
+      { rw cons_perm_iff_perm_erase,
+        refine ⟨mem_cons_of_mem _ hl.left, _⟩,
+        rw [erase_cons, if_neg hxy] },
+      refine this.symm.subperm.trans _,
+      rw subperm_cons,
+      refine IH _ hl.right,
+      rwa count_erase_of_ne hxy } }
+end
+
+theorem subperm_ext {l₁ l₂ : list α}
+  (h : ∀ x ∈ l₁, count x l₁ = count x l₂) : l₁ <+~ l₂ :=
```

Comment:
Isn't `\le` enough here?

Reply:
In which case, I'd kind of hope we already have this result since it would be useful when working with le on multisets.

Reply:
Changed it. As far as I can tell, there is no lemma about multisets connecting `count` to `le`. It only says in a docstring that
```
  Equivalently, `s ≤ t` means that `count a s ≤ count a t` for all `a`. -/
```
but no such lemma exists.